notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+   "metadata": {},
+   "source": [
+    "Okay, now we are ready to try an ingestion using the sortMerge join:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1bcc958-357c-4b64-b19f-bcbf0fc6941b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "REPLACE INTO \"wiki_merge\" OVERWRITE ALL\n",
+    "WITH \"wikidata\" AS \n",
+    "(\n",
+    "    SELECT *\n",
+    "    FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\n",
+    "          \"uris\":[ \"https://druid.apache.org/data/wikipedia.json.gz\"]\n",
+    "         }',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    ") EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR))\n",
+    "SELECT \n",
+    "  TIME_PARSE(d.\"timestamp\") as \"__time\",\n",
+    "  d.\"isRobot\", \n",
+    "  d.\"channel\" , \n",
+    "  d.\"timestamp\" , \n",
+    "  d.\"flags\" , \n",
+    "  d.\"isUnpatrolled\" , \n",
+    "  d.\"page\" , \n",
+    "  d.\"diffUrl\" , \n",
+    "  d.\"added\" , \n",
+    "  d.\"comment\" , \n",
+    "  d.\"commentLength\" , \n",
+    "  d.\"isNew\" , \n",
+    "  d.\"isMinor\" , \n",
+    "  d.\"delta\" , \n",
+    "  d.\"isAnonymous\" , \n",
+    "  d.\"user\" , \n",
+    "  d.\"deltaBucket\" , \n",
+    "  d.\"deleted\" , \n",
+    "  d.\"namespace\" , \n",
+    "  d.\"cityName\" , \n",
+    "  d.\"countryName\" , \n",
+    "  d.\"regionIsoCode\" , \n",
+    "  d.\"metroCode\" , \n",
+    "  d.\"countryIsoCode\" , \n",
+    "  d.\"regionName\", \n",
+    "  u.\"group\" AS \"user_group\",\n",
+    "  u.\"edits\" AS \"user_edits\",\n",
+    "  u.\"registered_at_ms\" AS \"user_registration_epoch\",\n",
+    "  u.\"language\" AS \"user_language\"\n",
+    "FROM \"wikidata\" AS d \n",
+    "  LEFT JOIN \"wiki_users\" AS u ON u.\"user\"=d.\"user\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "request = druid.sql.sql_request( sql)                 # init request object\n",
+    "request.add_context( 'sqlJoinAlgorithm', 'sortMerge') # use sortMerge to join the sources\n",
+    "request.add_context( 'maxNumTasks', 2)                # can't go any higher in test env\n",
+    "\n",
+    "druid.display.run_task(request)\n",
+    "druid.sql.wait_until_ready('wiki_merge')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "307374a4-635a-47bd-baf6-054373b3aa0c",
+   "metadata": {},
+   "source": [
+    "... and a final query to look at the newly joined data:"
```

Comment:
Run the next cell to query the newly joined data:

Reply:
done